### PR TITLE
fix: Add binding update to manual stat changes

### PR DESCRIPTION
**Pull request by mtolmacs**
Currently manual stats updates do not respect already bound linear elements.

### CODE REVIEW

**Comment by ryan-di** [comment]
What about the other direction, when it is the arrow (bound to some elements) that we're updating through stats panel? Should we update the bindings? For instance, the arrow is moved to a different location

image

**Comment by mtolmacs** [comment]
@dwelle Should we update the arrow bindings if manually update via the stats panel? It would make sense that if you move an arrow close to a bindable it would bind.

**Comment by mtolmacs** [comment]
I merged in the master branch and added in functionality to keep linear elements bound if they were previously bound and the manual changes don't remove them from their original binding area. Otherwise the linear element disconnects.

Previously it never disconnected the original binding, no matter how far it was moved via stats edits. Consequently it never bound to any shape either when moved within the binding area.

**Comment by dwelle** [comment]
> Should we update the arrow bindings if manually update via the stats panel? It would make sense that if you move an arrow close to a bindable it would bind.

> I merged in the master branch and added in functionality to keep linear elements bound if they were previously bound and the manual changes don't remove them from their original binding area. Otherwise the linear element disconnects.
>
> Previously it never disconnected the original binding, no matter how far it was moved via stats edits. Consequently it never bound to any shape either when moved within the binding area.

yeah, sounds good!

**Comment by mtolmacs** [comment]
Okay, if @ryan-di if you're okay with this, no more edge cases come to mind, then give me the go-ahead and I merge this.

**Comment by mtolmacs** [comment]
@ryan-di added tests and fixed a small bug. Let me know.

**Comment by mtolmacs** [comment]
Removed import reordering as per @dwelle's request.

**Comment by ryan-di** [comment]
Hey @mtolmacs, I've gone through the code and if you could take some time to go over the two comments, we're good to go 🙂

File: packages/excalidraw/components/Stats/Angle.tsx [hunk 1]
```diff
@@ -33,12 +37,19 @@ const handleDegreeChange: DragInputCallbackType<AngleProps["property"]> = ({
...
       return;
     }
+
     if (nextValue !== undefined) {
       const nextAngle = degreeToRadian(nextValue);
       mutateElement(latestElement, {
         angle: nextAngle,
       });
 
+      if (isLinearElement(latestElement)) {
```
[the remark this answers is not in the document]
Comment (mtolmacs) [reply packages/excalidraw/components/Stats/Angle.tsx]: Done!

**Review by mtolmacs — COMMENTED** [review]
Answered and completed

**Review by ryan-di — APPROVED** [review]
Thanks!
